### PR TITLE
8332243: Value classes implementing Cloneable

### PR DESCRIPTION
**Pull request by fparain**
Allow value classes to implement the Cloneable interface.

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8332243](https://bugs.openjdk.org/browse/JDK-8332243): Value classes implementing Cloneable (**Bug** - P3)

### Reviewers
 * [David Simms](https://openjdk.org/census#dsimms) (@MrSimms - Committer) ⚠️ Review applies to [55798abe](https://git.openjdk.org/valhalla/pull/1131/files/55798abe2401ee6ce34614c72b27a9b90a30a746)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/valhalla.git pull/1131/head:pull/1131` \
`$ git checkout pull/1131`

Update a local copy of the PR: \
`$ git checkout pull/1131` \
`$ git pull https://git.openjdk.org/valhalla.git pull/1131/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1131`

View PR using the GUI difftool: \
`$ git pr show -t 1131`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/valhalla/pull/1131.diff">https://git.openjdk.org/valhalla/pull/1131.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/valhalla/pull/1131#issuecomment-2168526667)

### CODE REVIEW

File: src/hotspot/share/prims/jvm.cpp [hunk 1]
```diff
@@ -720,13 +720,18 @@ JVM_ENTRY(jobject, JVM_Clone(JNIEnv* env, jobject handle))
...
-       klass->is_inline_klass() ||
       (klass->is_instance_klass() &&
        InstanceKlass::cast(klass)->reference_type() != REF_NONE)) {
     ResourceMark rm(THREAD);
     THROW_MSG_0(vmSymbols::java_lang_CloneNotSupportedException(), klass->external_name());
   }
 
+  if (klass->is_inline_klass()) {
+    // Value instances have no identity, so return the current instance instead of allocating a new one
+    // Value classes cannot have finalizers, so the method can return immediately
```
Comment (MrSimms) [inline src/hotspot/share/prims/jvm.cpp:731]: Nice

File: test/hotspot/jtreg/runtime/valhalla/inlinetypes/TestCloneableValue.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      double j;
+
+      public SimpleValue() {
+        i = 42;
+        j = Math.E;
+      }
+
+      @Override
+      public Object clone() throws CloneNotSupportedException {
+        return super.clone(); // deleguate to Object's method performing a shallow copy
```
Comment (MrSimms) [inline test/hotspot/jtreg/runtime/valhalla/inlinetypes/TestCloneableValue.java:51]: "delegate"

**Review by MrSimms — APPROVED** [review]
A tiny spelling thing, but LGTM

File: test/hotspot/jtreg/runtime/valhalla/inlinetypes/TestCloneableValue.java (hunk 2 above)
Comment (fparain) [reply test/hotspot/jtreg/runtime/valhalla/inlinetypes/TestCloneableValue.java]: Fixed

**Comment by fparain** [comment]
Thank you for the review

**Pull request closed without merging**